distributedlog-protocol/src/main/java/org/apache/distributedlog/EnvelopedRecordSetReader.java

```diff
@@ -115,4 +108,9 @@ public LogRecordWithDLSN nextRecord() throws IOException {
         return record;
     }
 
+    @Override
+    protected void finalize() throws Throwable {
```

Comment:
I know finalize is not a reliable and recommended way to release resource. However if I don't do finalize, there is no other way for me to release resources. or I have to change every classes that explicitly release the byte buf, this is going to a big change. also it is going to hard to change public structure like LogRecord to release references. it is going to break the API.

I will have a subsequent change to improve this. because I don't want to mix them in a big pull request.